src/brushes/pencil_brush.class.js

```diff
@@ -21,6 +35,37 @@
     initialize: function(canvas) {
       this.canvas = canvas;
       this._points = [];
+      this._attachKeyboardListeners();
+    },
+
+    /**
+     * Listens to `shift` key press
+     * @private
+     * @returns disposer
+     */
+    _attachKeyboardListeners: function () {
+      var _this = this;
+      var keyboardDown = function (e) {
+        if (e.shiftKey) {
+          e.preventDefault();
+          _this.drawStraightLine = true;
+        }
+      };
+      var keyboardUp = function () {
+        _this.drawStraightLine = false;
+      };
+
+      window.addEventListener('keydown', keyboardDown);
+      window.addEventListener('keyup', keyboardUp);
+      this._detach = function disposer() {
```

Comment:
Should we call `_detach` somewhere once the brush is dead? and if so from where?

Reply:
The event for SHIFT can be grabbed from the normal mouse event.
SHIFT, CTRL, ALT are modifier keys and can checked on the mousedown or mousemove.

Reply:
So you say no to attaching keyboard events.
Should I remove `_attachKeyboardListeners` entirely?

Reply:
ed43f43

Reply:
I have changed events to listen to canvas